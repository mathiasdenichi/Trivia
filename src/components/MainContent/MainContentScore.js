import React from 'react'
import PropTypes from 'prop-types'
import { FaCheckCircle, FaExclamationCircle } from 'react-icons/fa'

const Score = ({ correct, yourAnswer, question }) => (
  <tr>
    <td style={({ padding: '10px', width: '15%', textAlign: 'center' })} key={`${question}Correct`}>
      <p>{correct
        ? <FaCheckCircle style={({ color: 'limegreen', fontSize: '24px' })} />
        : <FaExclamationCircle style={({ color: 'orangered', fontSize: '24px' })} />}
      </p>
    </td>
    <td
      style={({ padding: '10px', width: '20%', textAlign: 'center', color: correct ? 'limegreen' : 'orangered' })}
      key={`${question}YourAnswer`}
    ><p>{yourAnswer}</p>
    </td>
    <td
      style={({ padding: '10px', width: '65%', color: correct ? 'limegreen' : 'orangered' })}
      key={`${question}Questions`}
    ><p>{question}</p>
    </td>
  </tr>
)

Score.propTypes = {
  correct: PropTypes.bool,
  yourAnswer: PropTypes.string,
  question: PropTypes.string,
}

Score.defaultProps = {
  correct: undefined,
  yourAnswer: '',
  question: '',
}

export default Score
