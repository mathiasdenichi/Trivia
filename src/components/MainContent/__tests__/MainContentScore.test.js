import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainContentScore from '../MainContentScore'

configure({ adapter: new Adapter() })

describe('Main Content Score', () => {
  const wrapper = shallow(<MainContentScore />)
  it('renders the component and matches it to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
