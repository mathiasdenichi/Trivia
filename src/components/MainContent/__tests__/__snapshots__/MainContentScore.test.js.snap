// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Content Score renders the component and matches it to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Score
    question=""
    yourAnswer=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          style={
            Object {
              "padding": "10px",
              "textAlign": "center",
              "width": "15%",
            }
          }
        >
          <p>
            <Unknown
              style={
                Object {
                  "color": "orangered",
                  "fontSize": "24px",
                }
              }
            />
          </p>
        </td>,
        <td
          style={
            Object {
              "color": "orangered",
              "padding": "10px",
              "textAlign": "center",
              "width": "20%",
            }
          }
        >
          <p>
            
          </p>
        </td>,
        <td
          style={
            Object {
              "color": "orangered",
              "padding": "10px",
              "width": "65%",
            }
          }
        >
          <p>
            
          </p>
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Correct",
        "nodeType": "host",
        "props": Object {
          "children": <p>
            <Unknown
              style={
                Object {
                  "color": "orangered",
                  "fontSize": "24px",
                }
              }
            />
          </p>,
          "style": Object {
            "padding": "10px",
            "textAlign": "center",
            "width": "15%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              style={
                Object {
                  "color": "orangered",
                  "fontSize": "24px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "color": "orangered",
                "fontSize": "24px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "p",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": "YourAnswer",
        "nodeType": "host",
        "props": Object {
          "children": <p>
            
          </p>,
          "style": Object {
            "color": "orangered",
            "padding": "10px",
            "textAlign": "center",
            "width": "20%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": "Questions",
        "nodeType": "host",
        "props": Object {
          "children": <p>
            
          </p>,
          "style": Object {
            "color": "orangered",
            "padding": "10px",
            "width": "65%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            style={
              Object {
                "padding": "10px",
                "textAlign": "center",
                "width": "15%",
              }
            }
          >
            <p>
              <Unknown
                style={
                  Object {
                    "color": "orangered",
                    "fontSize": "24px",
                  }
                }
              />
            </p>
          </td>,
          <td
            style={
              Object {
                "color": "orangered",
                "padding": "10px",
                "textAlign": "center",
                "width": "20%",
              }
            }
          >
            <p>
              
            </p>
          </td>,
          <td
            style={
              Object {
                "color": "orangered",
                "padding": "10px",
                "width": "65%",
              }
            }
          >
            <p>
              
            </p>
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Correct",
          "nodeType": "host",
          "props": Object {
            "children": <p>
              <Unknown
                style={
                  Object {
                    "color": "orangered",
                    "fontSize": "24px",
                  }
                }
              />
            </p>,
            "style": Object {
              "padding": "10px",
              "textAlign": "center",
              "width": "15%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Unknown
                style={
                  Object {
                    "color": "orangered",
                    "fontSize": "24px",
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "color": "orangered",
                  "fontSize": "24px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "p",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": "YourAnswer",
          "nodeType": "host",
          "props": Object {
            "children": <p>
              
            </p>,
            "style": Object {
              "color": "orangered",
              "padding": "10px",
              "textAlign": "center",
              "width": "20%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": "Questions",
          "nodeType": "host",
          "props": Object {
            "children": <p>
              
            </p>,
            "style": Object {
              "color": "orangered",
              "padding": "10px",
              "width": "65%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
